src/service/pinning/index.ts

```diff
@@ -64,26 +68,88 @@ async function placeOrderQueuedFiles() {
   logger.info('start placeOrderQueuedFiles');
   const pinObjects = await PinObjects.findAll({
     where: {
+      deleted: 0,
       [Op.or]: [
         {status: PinObjectStatus.failed},
         {status: PinObjectStatus.queued},
       ],
     },
   });
+  // distinct by cid
   if (_.isEmpty(pinObjects)) {
     logger.info('not pin objects to order');
     return;
   }
-  for (const obj of pinObjects) {
-    await placeOrderInCrust(obj.cid, obj.id).catch(e => {
-      logger.error(`order in crust failed: ${JSON.stringify(e)}`);
-    });
-    await sleep(configs.crust.orderTimeGap);
+  const cidList = _(pinObjects)
+    .map((i: any) => i.cid)
+    .uniq()
+    .value();
+  for (const cid of cidList) {
+    const needToOrder = await needOrder(cid);
```

Comment:
Can we pass the `pinObject` so that we don't need to query inside `needOrder` function?